v4/export/writer_util.go

```diff
@@ -114,6 +122,55 @@ func (b *writerPipe) ShouldSwitchStatement() bool {
 		(b.statementSizeLimit != UnspecifiedSize && b.currentStatementSize >= b.statementSizeLimit)
 }
 
+// WriteSpeedLimiter used for control speed of dump data
+type WriteSpeedLimiter interface {
+	CheckSpeed(*tcontext.Context, uint64)
+}
+
+// For no limit.
+type noWriteSpeedLimit struct{}
+
+func (sl *noWriteSpeedLimit) CheckSpeed(_ *tcontext.Context, _ uint64) {}
+
+// For limit.
+type rateWriteSpeedLimit struct {
+	limiter *rate.Limiter
+	limit   uint64
+}
+
+// The number of bytes currently downloaded has exceeded the speed limit value and needs to be paused.
+// Use a sliding window to prevent newSize from exceeding WaitN n.
+func (sl *rateWriteSpeedLimit) CheckSpeed(pCtx *tcontext.Context, newSize uint64) {
+	temp := newSize
+	for {
+		if pCtx == nil || pCtx.Err() != nil {
+			return
+		}
+		if temp <= sl.limit {
+			if err := sl.limiter.WaitN(pCtx, int(temp)); err != nil {
+				pCtx.L().Error("fail to write speed limit.", zap.Error(err))
+			}
+			break
+		}
+		if err := sl.limiter.WaitN(pCtx, int(sl.limit)); err != nil {
+			pCtx.L().Error("fail to write speed limit.", zap.Error(err))
+		}
+		temp -= sl.limit
+	}
+}
+
+// NewWriteSpeedLimiter is SpeedLimiter constructors build rateSpeedLimit or noSpeedLimit.
+func NewWriteSpeedLimiter(limit uint64) WriteSpeedLimiter {
+	if limit > 0 {
+		limiter := rate.NewLimiter(rate.Limit(limit), int(limit))
```

Comment:
no burst I think 🤔

Reply:
Sorry, Don't understand.

Burst what？

Reply:
the second parameter of `NewLimiter` is `burst`, do you expect that we should allow a burst?

Reply:
Do you mean don’t need parameter `burst`？

But the method `rate.NewLimiter` must be passed in two parameters.

Reply:
sorry I misunderstood your usage. I didn't notice you use WaitN rather than Wait. I have no problems with this line now.